src/main/java/frc/robot/Orchestrator.java

```diff
@@ -0,0 +1,287 @@
+        shooter.manualShoot(3.5).withTimeout(0.5),
+        Commands.parallel(indexer.setPercent(1), shooter.manualShoot(3.5))
+            .until(() -> !indexer.getLimitSwitchPressed())
+            .withTimeout(2));
+  }
+
+  /**
+   * Sets the arm to amp scoring position
+   *
+   * @return The command to move the arm to setpoint
+   */
+  public Command alignArmAmp() {
+    return arm.toSetpoint(ArmConstants.AMP_POSITION);
+  }
+
+  public Command goToAmp() {
+    Supplier<Pose2d> targetPose =
+        () ->
+            new Pose2d(
+                AllianceFlipUtil.apply(FieldConstants.ampCenter.getX()),
+                FieldConstants.ampCenter.getY() - 1,
+                new Rotation2d());
+    return Commands.defer(() -> new StraightDriveToPose(targetPose.get(), drive), Set.of(drive))
+        .andThen(
+            Commands.defer(
+                () ->
+                    new StraightDriveToPose(
+                        new Pose2d(
+                            drive.getPose().getTranslation().getX(),
+                            targetPose.get().getY() + 0.5,
+                            AllianceFlipUtil.apply(FieldConstants.ampCenter)
+                                .minus(drive.getPose().getTranslation())
+                                .getAngle()
+                                .minus(Rotation2d.fromRadians(Math.PI))),
+                        drive),
+                Set.of(drive)));
+  }
+
+  /**
+   * Uses goToAmp(), alignArmAmp(), and shootBasic() to move the robot to the amp and then line up
+   * and shoot.
+   *
+   * @return The command for scoring in the amp from any spot on the field.
+   */
+  public Command scoreAmp() {
+    return Commands.parallel(goToAmp(), alignArmAmp()).andThen(shootAmp());
+  }
+
+  /**
+   * Shoots with a velocity setpoint of 9999 rad per seconds, this needs to be tuned. Turns off the
+   * indexer and the shooter when the limit switch is not pressed.
+   *
+   * @return The command to shoot the note.
+   */
+  public Command shootBasic() {
+    return Commands.sequence(
+        shooter.manualShoot(10).withTimeout(1),
+        Commands.parallel(shooter.manualShoot(10), indexer.setPercent(1)).withTimeout(5));
+  }
+
+  /**
+   * Calculates the angle to align the arm to the speaker from anywhere on the field. Does arcTan of
+   * ((height of center of the speaker - height of the shooter) / distance to speaker)
+   *
+   * @return The command to move the arm to the correct setPoint for shooting from its current
+   *     location.
+   */
+  public Command alignArmSpeaker() {
+    return Commands.defer(
+        () ->
+            arm.toSetpoint(
+                new Rotation2d(
+                    Math.abs(
+                        Math.atan(
+                            (FieldConstants.Speaker.centerSpeakerOpening.getZ()
+                                    - Math.sin(arm.getAngle() - Units.degreesToRadians(13.95))
+                                        * Units.inchesToMeters(28))
```

Comment:
What's 13.95 degrees and 28 inches? move to constants

Reply:
This is abishek's code. IDK how it works and it isnt working rn. Has to be rewritten